sample_metadata/parser/cloudhelper.py

```diff
@@ -97,7 +97,12 @@ async def file_contents(self, filename) -> str | None:
 
     async def file_exists(self, filename: str) -> bool:
         """Determines whether a file exists"""
-        return AnyPath(self.file_path(filename)).exists()
+        path = self.file_path(filename)
+        if path.startswith('gs://'):
+            blob = self.get_gcs_blob(filename)
+            return blob is not None
```

Comment:
Couldn't you use https://cloud.google.com/python/docs/reference/storage/latest/google.cloud.storage.blob.Blob#google_cloud_storage_blob_Blob_exists instead?

As far as I remember, you can instantiate a blob directly (without calling `bucket.get_blob`, which you mention is problematic in `get_gcs_blob` below). That shouldn't require `get` permissions, I believe. Maybe that's also an opportunity to simplify `get_gcs_blob`, as long as callers don't assume that it includes an existence check.

Reply:
I've simplified `get_gcs_blob` to instantiate it directly.